### PR TITLE
refactor(generative_ai): Reduced hidden variables and refactored code (set-1)

### PR DESCRIPTION
**Pull request by Thoughtseize1**
| File 	| Comments 	|
|---	|---	|
| anthropic_claude_3_streaming.py 	| Updated PROJECT_ID variable. Hadrcode Location "us-east5" 	|
| anthropic_claude_3_tool_use.py 	| Updated PROJECT_ID variable. Hadrcode Location "us-east5" 	|
| anthropic_claude_3_unary.py 	| Updated PROJECT_ID variable. Hadrcode Location "us-east5" 	|
| chat.py 	| Just added if name == main 	|
| code_chat.py 	| Remove the temperature parameter 	|
| code_generation_function.py 	| Remove the temperature parameter 	|
| code_generation_unittest.py 	| Remove the temperature parameter 	|
| distillation.py 	| Made refactoring. Remove useless parametres. Update docstring 	|

## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [x] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: generative_ai/code_chat_test.py
Comment (msampathkumar) [inline generative_ai/code_chat_test.py:24]:
```suggestion
    assert len(content.text) > 0
```

**Review by msampathkumar — COMMENTED** [review]
Overall, LGTM.

**Comment by Thoughtseize1** [comment]
@msampathkumar We can merge it. Waiting for your approval

File: generative_ai/distillation.py
Comment (davidcavazos) [inline generative_ai/distillation.py:45]: Is passing the credentials explicitly like this required? Doesn't the `vertexai.init` function look for the default credentials if none is passed?

File: generative_ai/distillation.py
Comment (davidcavazos) [inline generative_ai/distillation.py:55]: Can the `teacher_model` be passed explicitly as well instead of relying on an "opaque variable"?

File: generative_ai/distillation.py
Comment (davidcavazos) [inline generative_ai/distillation.py:68]: It doesn't look like we have permissions to this `gs://bucket` bucket, or if it even exists. Would it be safe to either remove the `if __name__ == "__main__"` block? Or if you want it to be runnable, it should accept command line arguments to pass these in.

File: generative_ai/anthropic_claude_3_streaming.py
Comment (davidcavazos) [inline generative_ai/anthropic_claude_3_streaming.py:31]: Nit: can we have project_id go before region? It's a bit more consistent with other samples.

File: generative_ai/anthropic_claude_3_tool_use.py
Comment (davidcavazos) [inline generative_ai/anthropic_claude_3_tool_use.py:29]: Nit: same about project_id before region

Same with all other instances of this

File: generative_ai/anthropic_claude_3_streaming.py
Comment (davidcavazos) [inline generative_ai/anthropic_claude_3_streaming.py:17]: This will silently assign PROJECT_ID to be None. What error does the sample fail with if the `GOOGLE_CLOUD_PROJECT` environment variable is not set?

If we get a cryptic error, it might make more sense to `os.environ["GOOGLE_CLOUD_PROJECT"]`, that way we at least get an error right away if the variable is not set.

Personally, since this is a required argument, I would make the project be passed into the function and use `argparse` to get the project as an explicit required argument.

File: generative_ai/anthropic_claude_3_tool_use.py
Comment (davidcavazos) [inline generative_ai/anthropic_claude_3_tool_use.py:25]: Same around environment variables. Consider passing this as a function argument and doing some very simple `argparse` validations.

Same with all other instances of this

File: generative_ai/distillation.py
Comment (Thoughtseize1) [reply generative_ai/distillation.py:45]: I think, it should work without explicitly passing `credentials` 👍

File: generative_ai/distillation.py
Comment (Thoughtseize1) [reply generative_ai/distillation.py:55]: I changed the name of variable and added examples in docstring

File: generative_ai/distillation.py
Comment (Thoughtseize1) [reply generative_ai/distillation.py:68]: I am agree. I'ts unclear. I just rewrote all the arguments as examples.

File: generative_ai/anthropic_claude_3_streaming.py
Comment (Thoughtseize1) [reply generative_ai/anthropic_claude_3_streaming.py:31]: Done ✅

File: generative_ai/anthropic_claude_3_tool_use.py
Comment (Thoughtseize1) [reply generative_ai/anthropic_claude_3_tool_use.py:29]: Done ✅

File: generative_ai/anthropic_claude_3_streaming.py
Comment (Thoughtseize1) [reply generative_ai/anthropic_claude_3_streaming.py:17]: In this file on 27 line we ask developer to update `PROJECT_ID` variable
```
    # TODO(developer): Update and un-comment below line
    # PROJECT_ID = "PROJECT_ID"
```
Furthermore, the region tag begins with developer actions and does not display all the functions and parameters.
@davidcavazos

File: generative_ai/anthropic_claude_3_tool_use.py
Comment (Thoughtseize1) [reply generative_ai/anthropic_claude_3_tool_use.py:25]: I have replaced the region tag to simplify Sample and how it will look to our users.
PTAL

File: generative_ai/anthropic_claude_3_streaming.py
Comment (davidcavazos) [reply generative_ai/anthropic_claude_3_streaming.py:17]: Sounds good

**Review by davidcavazos — APPROVED** [review]

**Pull request merged**
